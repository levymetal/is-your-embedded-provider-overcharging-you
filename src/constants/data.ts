// https://www.esc.vic.gov.au/electricity-and-gas/prices-tariffs-and-benchmarks/victorian-default-offer

export const PRICING_DATA = {
  ausnet: {
    residential: {
      supply: 1.3354,
      usage: 0.3536, // block 1
      usage2: 0.3617, // block 2
      block1AnnualKWh: 1020 * 4, // up to 1020kWh used in a quarterly period
    },
    business: {
      supply: 1.3354,
      usage: 0.3837, // block 1
      usage2: 0.3945, // block 2
      block1AnnualKWh: 1020 * 4, // up to 1020kWh used in a quarterly period
    },
  },
  citipower: {
    residential: {
      supply: 1.1673,
      usage: 0.2575,
    },
    business: {
      supply: 1.3583,
      usage: 0.2529,
    },
  },
  jemena: {
    residential: {
      supply: 1.2105,
      usage: 0.3054,
    },
    business: {
      supply: 1.4055,
      usage: 0.3182,
    },
  },
  powercor: {
    residential: {
      supply: 1.3163,
      usage: 0.3045,
    },
    business: {
      supply: 1.4594,
      usage: 0.2797,
    },
  },
  united: {
    residential: {
      supply: 1.0955,
      usage: 0.2884,
    },
    business: {
      supply: 1.2865,
      usage: 0.2742,
    },
  },
};
