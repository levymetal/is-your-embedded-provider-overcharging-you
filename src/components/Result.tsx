import React from 'react';
import {css} from '@emotion/core';
import {PRICING_DATA} from '../constants/data';
import {format, getChargesIncGst, getRorts} from '../lib/helper';
import {colors} from '../styles/colors';
import Paragraph from './Paragraph';

type Props = {
  customerType: 'residential' | 'business';
  distributor: keyof typeof PRICING_DATA;
  supply: string;
  usage: string;
  usage2: string;
  gst?: string;
};

export default function Result({customerType, distributor, supply, usage, usage2, gst}: Props) {
  const isAusnet = distributor === 'ausnet';
  const prices = PRICING_DATA[distributor][customerType];
  const charges = getChargesIncGst(gst === 'inclusive', supply, usage, usage2);
  const rorts = getRorts(charges, prices);

  const message =
    rorts.supply && rorts.anyUsage
      ? 'You may be getting overcharged on both supply and usage.'
      : rorts.supply
      ? 'You may be getting overcharged on supply.'
      : rorts.anyUsage
      ? 'You may be getting overcharged on usage.'
      : 'Your rates seem to be acceptable.';

  return gst && (charges.supply || charges.usage || charges.usage2) ? (
    <section css={styles.root}>
      <div
        css={css`
          color: ${rorts.supply || rorts.anyUsage ? colors.vermilion : colors.oceanGreen};
        `}
      >
        <Emoji symbol={rorts.supply || rorts.anyUsage ? '😿' : '😸'} />
        <Paragraph>{message}</Paragraph>
      </div>
      <table css={styles.table}>
        <thead>
          <tr>
            <th aria-label="Charge" />
            <th>Your charge (inc gst)</th>
            <th>Maximum charge (inc gst)</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>Supply / day</td>
            <td css={rorts.supply ? styles.red : styles.green}>${format(charges.supply)}</td>
            <td>${format(prices.supply)}</td>
          </tr>
          <tr>
            <td>Usage / kWh {isAusnet && '(Block 1)'}</td>
            <td css={rorts.usage ? styles.red : styles.green}>${format(charges.usage)}</td>
            <td>${format(prices.usage)}</td>
          </tr>
          {isAusnet && (
            <tr>
              <td>Usage / kWh (Block 2)</td>
              <td css={rorts.usage2 ? styles.red : styles.green}>${format(charges.usage2)}</td>
              {/* @ts-ignore */}
              <td>${format(prices.usage2)}</td>
            </tr>
          )}
        </tbody>
      </table>
    </section>
  ) : (
    <section css={styles.root}>
      <Emoji symbol="🤷‍♀️" />
    </section>
  );
}

function Emoji({symbol}: {symbol: string}) {
  return (
    <span aria-hidden css={styles.emoji} role="img">
      {symbol}
    </span>
  );
}

const styles = {
  root: css`
    margin-top: 6rem;
    text-align: center;
  `,

  emoji: css`
    display: block;
    font-size: 8rem;
  `,

  table: css`
    font-size: 1.6rem;
    font-variant-numeric: tabular-nums;
    margin-top: 4rem;
    text-align: left;
    width: 100%;

    th,
    td:first-of-type {
      font-weight: bold;
    }

    td:first-of-type {
      width: 40%;
    }

    th,
    td {
      background: ${colors.alabaster};
      border: 0.1rem solid white;
      padding: 1em;
    }
  `,

  red: css`
    color: ${colors.vermilion};
  `,

  green: css`
    color: ${colors.oceanGreen};
  `,
};
