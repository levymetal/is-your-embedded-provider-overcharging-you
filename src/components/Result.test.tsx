import React from 'react';
import renderer from 'react-test-renderer';
import {PRICING_DATA} from '../constants/data';
import Result from './Result';

describe('Result', () => {
  const props = {
    customerType: 'residential',
    distributor: 'united',
    supply: String(PRICING_DATA.united.residential.supply),
    usage: String(PRICING_DATA.united.residential.usage),
    usage2: '',
    gst: 'inclusive',
  } as const;

  describe('default', () => {
    it('renders correctly', () => {
      expect(renderer.create(<Result {...props} />).toJSON()).toMatchSnapshot();
    });
  });

  describe('without charges', () => {
    it('renders correctly', () => {
      expect(renderer.create(<Result {...props} supply="" usage="" usage2="" />).toJSON()).toMatchSnapshot();
    });
  });

  describe('rorted on supply', () => {
    it('renders correctly', () => {
      const supply = String(Number(props.supply) + 0.05955);
      expect(renderer.create(<Result {...props} supply={supply} />).toJSON()).toMatchSnapshot();
    });
  });

  describe('rorted on supply but saving more on usage', () => {
    it('renders correctly', () => {
      const supply = String(Number(props.supply) + 0.05955);
      expect(renderer.create(<Result {...props} usage="0" supply={supply} />).toJSON()).toMatchSnapshot();
    });
  });

  describe('rorted on usage', () => {
    it('renders correctly', () => {
      const usage = String(Number(props.usage) + 0.02583);
      expect(renderer.create(<Result {...props} usage={usage} />).toJSON()).toMatchSnapshot();
    });
  });

  describe('rorted on usage but saving more on supply', () => {
    it('renders correctly', () => {
      const usage = String(Number(props.usage) + 0.02583);
      expect(renderer.create(<Result {...props} usage={usage} supply="0" />).toJSON()).toMatchSnapshot();
    });
  });

  describe('gst exclusive', () => {
    it('renders correctly', () => {
      expect(renderer.create(<Result {...props} gst="exclusive" />).toJSON()).toMatchSnapshot();
    });
  });

  describe('without gst', () => {
    it('renders correctly', () => {
      expect(renderer.create(<Result {...props} gst="" />).toJSON()).toMatchSnapshot();
    });
  });

  describe('ausnet', () => {
    it('renders correctly', () => {
      const {supply, usage, usage2} = PRICING_DATA.ausnet.residential;

      expect(
        renderer
          .create(
            <Result
              {...props}
              distributor="ausnet"
              supply={String(supply)}
              usage={String(usage)}
              usage2={String(usage2)}
            />
          )
          .toJSON()
      ).toMatchSnapshot();
    });
  });
});
