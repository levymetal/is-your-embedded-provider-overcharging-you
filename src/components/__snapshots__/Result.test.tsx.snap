// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result ausnet renders correctly 1`] = `
<section
  css={
    Object {
      "map": undefined,
      "name": "dift5x",
      "next": undefined,
      "styles": "
    margin-top: 6rem;
    text-align: center;
  ",
      "toString": [Function],
    }
  }
>
  <div
    css={
      Object {
        "map": undefined,
        "name": "1109p9j",
        "next": undefined,
        "styles": "
          color: #3cb371;
        ",
        "toString": [Function],
      }
    }
  >
    <span
      aria-hidden={true}
      css={
        Object {
          "map": undefined,
          "name": "1d2w5zj",
          "next": undefined,
          "styles": "
    display: block;
    font-size: 8rem;
  ",
          "toString": [Function],
        }
      }
      role="img"
    >
      😸
    </span>
    <p
      css={
        Object {
          "map": undefined,
          "name": "1l9jf5t",
          "next": undefined,
          "styles": "
        font-size: 1.8rem;
        line-height: 1.2;
        margin: 1em 0;
      ",
          "toString": [Function],
        }
      }
    >
      Your rates seem to be acceptable.
       
    </p>
  </div>
  <table
    css={
      Object {
        "map": undefined,
        "name": "li4c38",
        "next": undefined,
        "styles": "
    font-size: 1.6rem;
    font-variant-numeric: tabular-nums;
    margin-top: 4rem;
    text-align: left;
    width: 100%;

    th,
    td:first-of-type {
      font-weight: bold;
    }

    td:first-of-type {
      width: 40%;
    }

    th,
    td {
      background: #f9f9f9;
      border: 0.1rem solid white;
      padding: 1em;
    }
  ",
        "toString": [Function],
      }
    }
  >
    <thead>
      <tr>
        <th
          aria-label="Charge"
        />
        <th>
          Your charge (inc gst)
        </th>
        <th>
          Maximum charge (inc gst)
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Supply / day
        </td>
        <td
          css={
            Object {
              "map": undefined,
              "name": "xf633i",
              "next": undefined,
              "styles": "
    color: #3cb371;
  ",
              "toString": [Function],
            }
          }
        >
          $
          1.33540
        </td>
        <td>
          $
          1.33540
        </td>
      </tr>
      <tr>
        <td>
          Usage / kWh 
          (Block 1)
        </td>
        <td
          css={
            Object {
              "map": undefined,
              "name": "xf633i",
              "next": undefined,
              "styles": "
    color: #3cb371;
  ",
              "toString": [Function],
            }
          }
        >
          $
          0.35360
        </td>
        <td>
          $
          0.35360
        </td>
      </tr>
      <tr>
        <td>
          Usage / kWh (Block 2)
        </td>
        <td
          css={
            Object {
              "map": undefined,
              "name": "xf633i",
              "next": undefined,
              "styles": "
    color: #3cb371;
  ",
              "toString": [Function],
            }
          }
        >
          $
          0.36170
        </td>
        <td>
          $
          0.36170
        </td>
      </tr>
    </tbody>
  </table>
</section>
`;

exports[`Result default renders correctly 1`] = `
<section
  css={
    Object {
      "map": undefined,
      "name": "dift5x",
      "next": undefined,
      "styles": "
    margin-top: 6rem;
    text-align: center;
  ",
      "toString": [Function],
    }
  }
>
  <div
    css={
      Object {
        "map": undefined,
        "name": "1109p9j",
        "next": undefined,
        "styles": "
          color: #3cb371;
        ",
        "toString": [Function],
      }
    }
  >
    <span
      aria-hidden={true}
      css={
        Object {
          "map": undefined,
          "name": "1d2w5zj",
          "next": undefined,
          "styles": "
    display: block;
    font-size: 8rem;
  ",
          "toString": [Function],
        }
      }
      role="img"
    >
      😸
    </span>
    <p
      css={
        Object {
          "map": undefined,
          "name": "1l9jf5t",
          "next": undefined,
          "styles": "
        font-size: 1.8rem;
        line-height: 1.2;
        margin: 1em 0;
      ",
          "toString": [Function],
        }
      }
    >
      Your rates seem to be acceptable.
       
    </p>
  </div>
  <table
    css={
      Object {
        "map": undefined,
        "name": "li4c38",
        "next": undefined,
        "styles": "
    font-size: 1.6rem;
    font-variant-numeric: tabular-nums;
    margin-top: 4rem;
    text-align: left;
    width: 100%;

    th,
    td:first-of-type {
      font-weight: bold;
    }

    td:first-of-type {
      width: 40%;
    }

    th,
    td {
      background: #f9f9f9;
      border: 0.1rem solid white;
      padding: 1em;
    }
  ",
        "toString": [Function],
      }
    }
  >
    <thead>
      <tr>
        <th
          aria-label="Charge"
        />
        <th>
          Your charge (inc gst)
        </th>
        <th>
          Maximum charge (inc gst)
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Supply / day
        </td>
        <td
          css={
            Object {
              "map": undefined,
              "name": "xf633i",
              "next": undefined,
              "styles": "
    color: #3cb371;
  ",
              "toString": [Function],
            }
          }
        >
          $
          1.09550
        </td>
        <td>
          $
          1.09550
        </td>
      </tr>
      <tr>
        <td>
          Usage / kWh 
        </td>
        <td
          css={
            Object {
              "map": undefined,
              "name": "xf633i",
              "next": undefined,
              "styles": "
    color: #3cb371;
  ",
              "toString": [Function],
            }
          }
        >
          $
          0.28840
        </td>
        <td>
          $
          0.28840
        </td>
      </tr>
    </tbody>
  </table>
</section>
`;

exports[`Result gst exclusive renders correctly 1`] = `
<section
  css={
    Object {
      "map": undefined,
      "name": "dift5x",
      "next": undefined,
      "styles": "
    margin-top: 6rem;
    text-align: center;
  ",
      "toString": [Function],
    }
  }
>
  <div
    css={
      Object {
        "map": undefined,
        "name": "h7vb22",
        "next": undefined,
        "styles": "
          color: #ff4500;
        ",
        "toString": [Function],
      }
    }
  >
    <span
      aria-hidden={true}
      css={
        Object {
          "map": undefined,
          "name": "1d2w5zj",
          "next": undefined,
          "styles": "
    display: block;
    font-size: 8rem;
  ",
          "toString": [Function],
        }
      }
      role="img"
    >
      😿
    </span>
    <p
      css={
        Object {
          "map": undefined,
          "name": "1l9jf5t",
          "next": undefined,
          "styles": "
        font-size: 1.8rem;
        line-height: 1.2;
        margin: 1em 0;
      ",
          "toString": [Function],
        }
      }
    >
      You may be getting overcharged on both supply and usage.
       
      This could be costing you around $155 a year.
    </p>
  </div>
  <table
    css={
      Object {
        "map": undefined,
        "name": "li4c38",
        "next": undefined,
        "styles": "
    font-size: 1.6rem;
    font-variant-numeric: tabular-nums;
    margin-top: 4rem;
    text-align: left;
    width: 100%;

    th,
    td:first-of-type {
      font-weight: bold;
    }

    td:first-of-type {
      width: 40%;
    }

    th,
    td {
      background: #f9f9f9;
      border: 0.1rem solid white;
      padding: 1em;
    }
  ",
        "toString": [Function],
      }
    }
  >
    <thead>
      <tr>
        <th
          aria-label="Charge"
        />
        <th>
          Your charge (inc gst)
        </th>
        <th>
          Maximum charge (inc gst)
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Supply / day
        </td>
        <td
          css={
            Object {
              "map": undefined,
              "name": "1xs99n",
              "next": undefined,
              "styles": "
    color: #ff4500;
  ",
              "toString": [Function],
            }
          }
        >
          $
          1.20505
        </td>
        <td>
          $
          1.09550
        </td>
      </tr>
      <tr>
        <td>
          Usage / kWh 
        </td>
        <td
          css={
            Object {
              "map": undefined,
              "name": "1xs99n",
              "next": undefined,
              "styles": "
    color: #ff4500;
  ",
              "toString": [Function],
            }
          }
        >
          $
          0.31724
        </td>
        <td>
          $
          0.28840
        </td>
      </tr>
    </tbody>
  </table>
</section>
`;

exports[`Result rorted on supply but saving more on usage renders correctly 1`] = `
<section
  css={
    Object {
      "map": undefined,
      "name": "dift5x",
      "next": undefined,
      "styles": "
    margin-top: 6rem;
    text-align: center;
  ",
      "toString": [Function],
    }
  }
>
  <div
    css={
      Object {
        "map": undefined,
        "name": "1109p9j",
        "next": undefined,
        "styles": "
          color: #3cb371;
        ",
        "toString": [Function],
      }
    }
  >
    <span
      aria-hidden={true}
      css={
        Object {
          "map": undefined,
          "name": "1d2w5zj",
          "next": undefined,
          "styles": "
    display: block;
    font-size: 8rem;
  ",
          "toString": [Function],
        }
      }
      role="img"
    >
      😸
    </span>
    <p
      css={
        Object {
          "map": undefined,
          "name": "1l9jf5t",
          "next": undefined,
          "styles": "
        font-size: 1.8rem;
        line-height: 1.2;
        margin: 1em 0;
      ",
          "toString": [Function],
        }
      }
    >
      You may be getting overcharged on supply, but your usage charge looks good.
       
      You’re saving around $1132 per year compared to the Victorian Default Offer.
    </p>
  </div>
  <table
    css={
      Object {
        "map": undefined,
        "name": "li4c38",
        "next": undefined,
        "styles": "
    font-size: 1.6rem;
    font-variant-numeric: tabular-nums;
    margin-top: 4rem;
    text-align: left;
    width: 100%;

    th,
    td:first-of-type {
      font-weight: bold;
    }

    td:first-of-type {
      width: 40%;
    }

    th,
    td {
      background: #f9f9f9;
      border: 0.1rem solid white;
      padding: 1em;
    }
  ",
        "toString": [Function],
      }
    }
  >
    <thead>
      <tr>
        <th
          aria-label="Charge"
        />
        <th>
          Your charge (inc gst)
        </th>
        <th>
          Maximum charge (inc gst)
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Supply / day
        </td>
        <td
          css={
            Object {
              "map": undefined,
              "name": "1xs99n",
              "next": undefined,
              "styles": "
    color: #ff4500;
  ",
              "toString": [Function],
            }
          }
        >
          $
          1.15505
        </td>
        <td>
          $
          1.09550
        </td>
      </tr>
      <tr>
        <td>
          Usage / kWh 
        </td>
        <td
          css={
            Object {
              "map": undefined,
              "name": "xf633i",
              "next": undefined,
              "styles": "
    color: #3cb371;
  ",
              "toString": [Function],
            }
          }
        >
          $
          0.00000
        </td>
        <td>
          $
          0.28840
        </td>
      </tr>
    </tbody>
  </table>
</section>
`;

exports[`Result rorted on supply renders correctly 1`] = `
<section
  css={
    Object {
      "map": undefined,
      "name": "dift5x",
      "next": undefined,
      "styles": "
    margin-top: 6rem;
    text-align: center;
  ",
      "toString": [Function],
    }
  }
>
  <div
    css={
      Object {
        "map": undefined,
        "name": "h7vb22",
        "next": undefined,
        "styles": "
          color: #ff4500;
        ",
        "toString": [Function],
      }
    }
  >
    <span
      aria-hidden={true}
      css={
        Object {
          "map": undefined,
          "name": "1d2w5zj",
          "next": undefined,
          "styles": "
    display: block;
    font-size: 8rem;
  ",
          "toString": [Function],
        }
      }
      role="img"
    >
      😿
    </span>
    <p
      css={
        Object {
          "map": undefined,
          "name": "1l9jf5t",
          "next": undefined,
          "styles": "
        font-size: 1.8rem;
        line-height: 1.2;
        margin: 1em 0;
      ",
          "toString": [Function],
        }
      }
    >
      You may be getting overcharged on supply.
       
      This could be costing you around $22 a year.
    </p>
  </div>
  <table
    css={
      Object {
        "map": undefined,
        "name": "li4c38",
        "next": undefined,
        "styles": "
    font-size: 1.6rem;
    font-variant-numeric: tabular-nums;
    margin-top: 4rem;
    text-align: left;
    width: 100%;

    th,
    td:first-of-type {
      font-weight: bold;
    }

    td:first-of-type {
      width: 40%;
    }

    th,
    td {
      background: #f9f9f9;
      border: 0.1rem solid white;
      padding: 1em;
    }
  ",
        "toString": [Function],
      }
    }
  >
    <thead>
      <tr>
        <th
          aria-label="Charge"
        />
        <th>
          Your charge (inc gst)
        </th>
        <th>
          Maximum charge (inc gst)
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Supply / day
        </td>
        <td
          css={
            Object {
              "map": undefined,
              "name": "1xs99n",
              "next": undefined,
              "styles": "
    color: #ff4500;
  ",
              "toString": [Function],
            }
          }
        >
          $
          1.15505
        </td>
        <td>
          $
          1.09550
        </td>
      </tr>
      <tr>
        <td>
          Usage / kWh 
        </td>
        <td
          css={
            Object {
              "map": undefined,
              "name": "xf633i",
              "next": undefined,
              "styles": "
    color: #3cb371;
  ",
              "toString": [Function],
            }
          }
        >
          $
          0.28840
        </td>
        <td>
          $
          0.28840
        </td>
      </tr>
    </tbody>
  </table>
</section>
`;

exports[`Result rorted on usage but saving more on supply renders correctly 1`] = `
<section
  css={
    Object {
      "map": undefined,
      "name": "dift5x",
      "next": undefined,
      "styles": "
    margin-top: 6rem;
    text-align: center;
  ",
      "toString": [Function],
    }
  }
>
  <div
    css={
      Object {
        "map": undefined,
        "name": "1109p9j",
        "next": undefined,
        "styles": "
          color: #3cb371;
        ",
        "toString": [Function],
      }
    }
  >
    <span
      aria-hidden={true}
      css={
        Object {
          "map": undefined,
          "name": "1d2w5zj",
          "next": undefined,
          "styles": "
    display: block;
    font-size: 8rem;
  ",
          "toString": [Function],
        }
      }
      role="img"
    >
      😸
    </span>
    <p
      css={
        Object {
          "map": undefined,
          "name": "1l9jf5t",
          "next": undefined,
          "styles": "
        font-size: 1.8rem;
        line-height: 1.2;
        margin: 1em 0;
      ",
          "toString": [Function],
        }
      }
    >
      You may be getting overcharged on usage, but your supply charge looks good.
       
      You’re saving around $297 per year compared to the Victorian Default Offer.
    </p>
  </div>
  <table
    css={
      Object {
        "map": undefined,
        "name": "li4c38",
        "next": undefined,
        "styles": "
    font-size: 1.6rem;
    font-variant-numeric: tabular-nums;
    margin-top: 4rem;
    text-align: left;
    width: 100%;

    th,
    td:first-of-type {
      font-weight: bold;
    }

    td:first-of-type {
      width: 40%;
    }

    th,
    td {
      background: #f9f9f9;
      border: 0.1rem solid white;
      padding: 1em;
    }
  ",
        "toString": [Function],
      }
    }
  >
    <thead>
      <tr>
        <th
          aria-label="Charge"
        />
        <th>
          Your charge (inc gst)
        </th>
        <th>
          Maximum charge (inc gst)
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Supply / day
        </td>
        <td
          css={
            Object {
              "map": undefined,
              "name": "xf633i",
              "next": undefined,
              "styles": "
    color: #3cb371;
  ",
              "toString": [Function],
            }
          }
        >
          $
          0.00000
        </td>
        <td>
          $
          1.09550
        </td>
      </tr>
      <tr>
        <td>
          Usage / kWh 
        </td>
        <td
          css={
            Object {
              "map": undefined,
              "name": "1xs99n",
              "next": undefined,
              "styles": "
    color: #ff4500;
  ",
              "toString": [Function],
            }
          }
        >
          $
          0.31423
        </td>
        <td>
          $
          0.28840
        </td>
      </tr>
    </tbody>
  </table>
</section>
`;

exports[`Result rorted on usage renders correctly 1`] = `
<section
  css={
    Object {
      "map": undefined,
      "name": "dift5x",
      "next": undefined,
      "styles": "
    margin-top: 6rem;
    text-align: center;
  ",
      "toString": [Function],
    }
  }
>
  <div
    css={
      Object {
        "map": undefined,
        "name": "h7vb22",
        "next": undefined,
        "styles": "
          color: #ff4500;
        ",
        "toString": [Function],
      }
    }
  >
    <span
      aria-hidden={true}
      css={
        Object {
          "map": undefined,
          "name": "1d2w5zj",
          "next": undefined,
          "styles": "
    display: block;
    font-size: 8rem;
  ",
          "toString": [Function],
        }
      }
      role="img"
    >
      😿
    </span>
    <p
      css={
        Object {
          "map": undefined,
          "name": "1l9jf5t",
          "next": undefined,
          "styles": "
        font-size: 1.8rem;
        line-height: 1.2;
        margin: 1em 0;
      ",
          "toString": [Function],
        }
      }
    >
      You may be getting overcharged on usage.
       
      This could be costing you around $103 a year.
    </p>
  </div>
  <table
    css={
      Object {
        "map": undefined,
        "name": "li4c38",
        "next": undefined,
        "styles": "
    font-size: 1.6rem;
    font-variant-numeric: tabular-nums;
    margin-top: 4rem;
    text-align: left;
    width: 100%;

    th,
    td:first-of-type {
      font-weight: bold;
    }

    td:first-of-type {
      width: 40%;
    }

    th,
    td {
      background: #f9f9f9;
      border: 0.1rem solid white;
      padding: 1em;
    }
  ",
        "toString": [Function],
      }
    }
  >
    <thead>
      <tr>
        <th
          aria-label="Charge"
        />
        <th>
          Your charge (inc gst)
        </th>
        <th>
          Maximum charge (inc gst)
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Supply / day
        </td>
        <td
          css={
            Object {
              "map": undefined,
              "name": "xf633i",
              "next": undefined,
              "styles": "
    color: #3cb371;
  ",
              "toString": [Function],
            }
          }
        >
          $
          1.09550
        </td>
        <td>
          $
          1.09550
        </td>
      </tr>
      <tr>
        <td>
          Usage / kWh 
        </td>
        <td
          css={
            Object {
              "map": undefined,
              "name": "1xs99n",
              "next": undefined,
              "styles": "
    color: #ff4500;
  ",
              "toString": [Function],
            }
          }
        >
          $
          0.31423
        </td>
        <td>
          $
          0.28840
        </td>
      </tr>
    </tbody>
  </table>
</section>
`;

exports[`Result without charges renders correctly 1`] = `
<section
  css={
    Object {
      "map": undefined,
      "name": "dift5x",
      "next": undefined,
      "styles": "
    margin-top: 6rem;
    text-align: center;
  ",
      "toString": [Function],
    }
  }
>
  <span
    aria-hidden={true}
    css={
      Object {
        "map": undefined,
        "name": "1d2w5zj",
        "next": undefined,
        "styles": "
    display: block;
    font-size: 8rem;
  ",
        "toString": [Function],
      }
    }
    role="img"
  >
    🤷‍♀️
  </span>
</section>
`;

exports[`Result without gst renders correctly 1`] = `
<section
  css={
    Object {
      "map": undefined,
      "name": "dift5x",
      "next": undefined,
      "styles": "
    margin-top: 6rem;
    text-align: center;
  ",
      "toString": [Function],
    }
  }
>
  <span
    aria-hidden={true}
    css={
      Object {
        "map": undefined,
        "name": "1d2w5zj",
        "next": undefined,
        "styles": "
    display: block;
    font-size: 8rem;
  ",
        "toString": [Function],
      }
    }
    role="img"
  >
    🤷‍♀️
  </span>
</section>
`;
